include/core/shape.h

```diff
@@ -378,6 +378,178 @@ class Sphere : public Shape<dim>
 #endif
 };
 
+/**
+ * @class This class defines superquadric shapes. Their signed distance function
+ * is:
+ * |\frac{x}{a}|^r  + |\frac{y}{b}|^s + |\frac{z}{c}|^t - 1 = 0
+ * @tparam dim Dimension of the shape
+ */
+template <int dim>
+class Superquadric : public Shape<dim>
+{
+public:
+  /**
+   * @brief Constructor for a superquadric shape
+   * @param half_lengths The half-lengths of each direction
+   * @param exponents The blockiness in each direction
+   * @param position The superquadric center
+   * @param orientation The superquadric orientation
+   */
+  Superquadric<dim>(const Tensor<1, dim> half_lengths,
+                    const Tensor<1, dim> exponents,
+                    const Point<dim> &   position,
+                    const Tensor<1, 3> & orientation)
+    : Shape<dim>(half_lengths.norm(), position, orientation)
+  {
+    this->half_lengths = half_lengths;
+    this->exponents    = exponents;
+    this->epsilon      = 1e-12;
```

Comment:
Epsilon is hardcoded to a very small value at the moment. I think this should be a parameter

Reply:
I thought about this also, but I wasn't sure if it would be useful. I can add it as a 7th parameter.